commit efd00b8ff05cb78ecb0351b96cce7780bcb72a2a
Author: David Carlier <devnexen@gmail.com>
Date:   Tue Jul 16 19:55:04 2024 +0100

    ext/curl: curl_error using curl_easy_strerror if CURLOPT_ERRORBUFFER
    
    did not fill the error buffer.
    
    close GH-14984

diff --git a/ext/curl/interface.c b/ext/curl/interface.c
index 707f4e0a6f..4884ddc822 100644
--- a/ext/curl/interface.c
+++ b/ext/curl/interface.c
@@ -2764,7 +2764,11 @@ PHP_FUNCTION(curl_error)
 
 	if (ch->err.no) {
 		ch->err.str[CURL_ERROR_SIZE] = 0;
-		RETURN_STRING(ch->err.str);
+		if (strlen(ch->err.str) > 0) {
+			RETURN_STRING(ch->err.str);
+		} else {
+			RETURN_STRING(curl_easy_strerror(ch->err.no));
+		}
 	} else {
 		RETURN_EMPTY_STRING();
 	}
